iree/compiler/Codegen/LLVMCPU/LLVMCPUTileAndVectorizeLinalgTensorOps2.cpp

```diff
@@ -0,0 +1,333 @@
+#include "iree/compiler/Codegen/LLVMCPU/KernelDispatch.h"
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "iree/compiler/Codegen/Transforms/Transforms.h"
+#include "iree/compiler/Codegen/Utils/MarkerUtils.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "mlir/Conversion/VectorToSCF/VectorToSCF.h"
+#include "mlir/Dialect/Linalg/IR/LinalgOps.h"
+#include "mlir/Dialect/Linalg/Transforms/Hoisting.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/MemRef/Transforms/Passes.h"
+#include "mlir/Dialect/SCF/Transforms.h"
+#include "mlir/Dialect/Vector/VectorTransforms.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-llvmcpu-tile-and-vectorize"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+// Could just be linalg::TilingPattern with a ContractionOpInterface filter, but
+// that is always templated on an op.
+struct TileWorkgroups : public linalg::LinalgBaseTilingPattern {
+  using Base = linalg::LinalgBaseTilingPattern;
+  TileWorkgroups(MLIRContext *context, linalg::LinalgTilingOptions options,
+                 linalg::LinalgTransformationFilter marker)
+      : LinalgBaseTilingPattern(context, options, marker) {}
+  LogicalResult matchAndRewrite(Operation *op,
+                                PatternRewriter &rewriter) const override {
+    auto contractionOp = dyn_cast<linalg::ContractionOpInterface>(op);
+    if (!contractionOp) return failure();
+
+    linalg::TiledLinalgOp tiledLinalgOp;
+    if (failed(Base::matchAndRewriteBase(op, rewriter, tiledLinalgOp))) {
+      return failure();
+    }
+    rewriter.replaceOp(op, tiledLinalgOp.tensorResults);
+    return success();
+  }
+};
+
+}  // namespace
+
+namespace {
+struct LLVMCPUTileAndVectorizePass2
+    : public LLVMCPUTileAndVectorize2Base<LLVMCPUTileAndVectorizePass2> {
+  LLVMCPUTileAndVectorizePass2(bool vectorize = true)
+      : lowerToVectors(vectorize) {}
+  LLVMCPUTileAndVectorizePass2(const LLVMCPUTileAndVectorizePass2 &pass) {
+    lowerToVectors = pass.lowerToVectors;
+  }
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<linalg::LinalgDialect, memref::MemRefDialect,
+                    vector::VectorDialect>();
+  }
+  void runOnOperation() override;
+
+ private:
+  bool lowerToVectors;
+};
+}  // namespace
+
+void LLVMCPUTileAndVectorizePass2::runOnOperation() {
+  MLIRContext *context = &getContext();
+  auto funcOp = getOperation();
+
+  DEBUG_WITH_TYPE(DEBUG_TYPE, {
+    llvm::dbgs() << "\n--- Before LLVMCPUTileAndVectorizePass2 ---\n";
+    funcOp.print(llvm::dbgs(), OpPrintingFlags().useLocalScope());
+    llvm::dbgs() << "\n\n";
+  });
+
+  // Assume there is a single op with a lowering config we use to drive the
+  // tiling decisions.
+  IREE::Codegen::LoweringConfigAttr config;
```

Comment:
I think we should be able to drop this. Lets assume for now that we are always going to do L1 tiling (we will have a separate pipeline for just one level tile + vectorize)

Reply:
I don't get it. Shouldn't we need this to get tiling sizes information?

Reply:
Got you, thanks!